decorum/decorum.py

```diff
@@ -26,40 +29,40 @@ def __init__(self, *args, **kwargs):
         #: directly to the matching attributes on the decorator.
         self.assigned = functools.WRAPPER_ASSIGNMENTS
         if 'assigned' in kwargs:
-            self.assigned = kwargs['assigned']
+            self.assigned = kwargs.pop('assigned')
 
         if args and callable(args[0]):
             # used as decorator without being called
             self.init()
-            self._wrapped = self.__call__(args[0])
+            self.wraps(args[0])
         else:
             # used as decorator and called
             self.init(*args, **kwargs)
 
-    def __call__(self, f=None, *args, **kwargs):
+    def __call__(self, *args, **kwargs):
         """Actually run the decorated function.
 
         Uses :meth:`wrap` to handle decoration. Restores :attr:`__doc__` and
         :attr:`__name__`.
 
         """
-        if not callable(f):
-            if f:
-                return self._wrapped(f, *args, **kwargs)
-            else:
-                return self._wrapped(*args, **kwargs)
+        if self._wrapped:
+            return self.call(*args, **kwargs)
         else:
-            wrapped = self.wraps(f)
-            return wrapped
+            f = args[0]
+            return self.wraps(f)
 
     def wraps(self, f):
         """Wraps the function and returns it"""
+        self._wrapped = f
         functools.update_wrapper(self, f, self.assigned or (), ())
         return self
 
-    def init(self, *args, **kwargs):
+    def init(self):
```

Comment:
I think `init()` signature should tell exactly the decorator's own options.
I mean:
* if the decorator doesn't accept configuration, then `def init(self):`.
* if the decorator requires one positional argument, then `def init(self, this_is_required):`.
* if the decorator accepts one argument with a default value, then `def init(self, this_is_optional=True):`.
* if the decorator accepts only keyword arguments, then `def init(self, **kwargs):`
* ...